model/search_params_test.go

```diff
@@ -12,924 +12,1500 @@ import (
-	require.Len(t, words, 3)
-	require.Equal(t, "apple", words[0].value)
-	require.False(t, words[0].exclude)
-	require.Equal(t, "banana", words[1].value)
-	require.False(t, words[1].exclude)
-	require.Equal(t, "#fruit", words[2].value)
-	require.False(t, words[2].exclude)
-	require.Len(t, flags, 2)
-	require.Equal(t, "after", flags[0].name)
-	require.Equal(t, "2018-1-1", flags[0].value)
-	require.False(t, flags[0].exclude)
-	require.Equal(t, "before", flags[1].name)
-	require.Equal(t, "2018-1-10", flags[1].value)
-	require.False(t, flags[1].exclude)
-
-	words, flags = parseSearchFlags(splitWords("test after:2018-7-1"))
-	require.Len(t, words, 1)
-	require.Equal(t, "test", words[0].value)
-	require.False(t, words[0].exclude)
-	require.Len(t, flags, 1)
-	require.Equal(t, "after", flags[0].name)
-	require.Equal(t, "2018-7-1", flags[0].value)
-	require.False(t, flags[0].exclude)
-
-	words, flags = parseSearchFlags(splitWords("test on:2018-7-1"))
-	require.Len(t, words, 1)
-	require.Equal(t, "test", words[0].value)
-	require.False(t, words[0].exclude)
-	require.Len(t, flags, 1)
-	require.Equal(t, "on", flags[0].name)
-	require.Equal(t, "2018-7-1", flags[0].value)
-	require.False(t, flags[0].exclude)
+func TestParseSearchFlags2(t *testing.T) {
+	for _, testCase := range []struct {
+		Name  string
+		Input string
 
-	words, flags = parseSearchFlags(splitWords("-on:2018-7-1 test"))
-	require.Len(t, words, 1)
-	require.Equal(t, "test", words[0].value)
-	require.False(t, words[0].exclude)
-	require.Len(t, flags, 1)
-	require.Equal(t, "on", flags[0].name)
-	require.Equal(t, "2018-7-1", flags[0].value)
-	require.True(t, flags[0].exclude)
+		Words []searchWord
+		Flags []flag
+	}{
+		{
+			Name:  "string is empty",
+			Input: "",
+			Words: []searchWord{},
+			Flags: []flag{},
+		},
+		{
+			Name:  "string is a single word",
+			Input: "word",
+			Words: []searchWord{
+				{
+					value:   "word",
+					exclude: false,
+				},
+			},
+			Flags: []flag{},
+		},
+		{
+			Name:  "string is a single word with a - prefix",
+			Input: "-word",
+			Words: []searchWord{
+				{
+					value:   "word",
+					exclude: true,
+				},
+			},
+			Flags: []flag{},
+		},
+		{
+			Name:  "string is multiple words all with - prefix",
+			Input: "-apple -banana -cherry",
+			Words: []searchWord{
+				{
+					value:   "apple",
+					exclude: true,
+				},
+				{
+					value:   "banana",
+					exclude: true,
+				},
+				{
+					value:   "cherry",
+					exclude: true,
+				},
+			},
+			Flags: []flag{},
+		},
+		{
+			Name:  "string is multiple words with a single - prefix",
+			Input: "apple -banana cherry",
+			Words: []searchWord{
+				{
+					value:   "apple",
+					exclude: false,
+				},
+				{
+					value:   "banana",
+					exclude: true,
+				},
+				{
+					value:   "cherry",
+					exclude: false,
+				},
+			},
+			Flags: []flag{},
+		},
+		{
+			Name:  "string is multiple words containing a :",
```

Comment:
instead of `containing a :` I would use `containing a flag`, and apply this same change to all the subsequent tests